code/modules/projectiles/guns/energy/hos.dm

```diff
@@ -0,0 +1,304 @@
+	var/max_shots = 6
+	var/type_color = null
+	var/type_name = null
+
+	var/obj/item/stock_parts/cell/cell
+	var/cell_type = /obj/item/stock_parts/cell{charge = 600; maxcharge = 600}
+
+	var/e_cost = 100
+	projectile_type = /obj/item/projectile/beam
+
+/obj/item/ammo_casing/mws_batt/Initialize()
+	. = ..()
+	pixel_x = rand(-10, 10)
+	pixel_y = rand(-10, 10)
+	cell = new cell_type(src)
+	cell.give(cell.maxcharge)
+	BB = null
+	update_icon()
+
+/obj/item/ammo_casing/mws_batt/update_icon()
+	cut_overlays()
+
+	var/image/ends = image(icon, icon_state = "[initial(icon_state)]_ends")
+	ends.color = type_color
+	add_overlay(ends)
+
+// Specific batteries //
+/obj/item/ammo_casing/mws_batt/lethal
+	name = "\'MWS\' microbattery - LETHAL"
+	type_color = "#bf3d3d"
+	type_name = "<span style='color:#bf3d3d;font-weight:bold;'>LETHAL</span>"
+	projectile_type = /obj/item/projectile/beam
+
+/obj/item/ammo_casing/mws_batt/stun
+	name = "\'MWS\' microbattery - STUN"
+	type_color = "#0f81bc"
+	type_name = "<span style='color:#0f81bc;font-weight:bold;'>STUN</span>"
+	projectile_type = /obj/item/projectile/beam/disabler
+
+/obj/item/ammo_casing/mws_batt/net
+	name = "\'MWS\' microbattery - NET"
+	type_color = "#43f136"
+	type_name = "<span style='color:#43d136;font-weight:bold;'>NET</span>"
+	projectile_type = /obj/item/projectile/energy/net
+
+/obj/item/ammo_casing/mws_batt/xray
+	name = "\'MWS\' microbattery - XRAY"
+	type_color = "#32c025"
+	type_name = "<span style='color:#32c025;font-weight:bold;'>XRAY</span>"
+	projectile_type = /obj/item/projectile/beam/xray
+
+
+/obj/item/ammo_casing/mws_batt/ion
+	name = "\'MWS\' microbattery - ION"
+	type_color = "#d084d6"
+	type_name = "<span style='color:#d084d6;font-weight:bold;'>ION</span>"
+	projectile_type = /obj/item/projectile/ion
+
+
+/obj/item/ammo_casing/mws_batt/final
+	name = "\'mws\' microbattery - FINAL OPTION"
+	type_color = "#fcfc0f"
+	type_name = "<span style='color:#000000;font-weight:bold;'>FINAL OPTION</span>" //Doesn't look good in yellow in chat
+	projectile_type = /obj/item/projectile/beam/final_option
+
+/obj/item/projectile/beam/final_option
+	name = "final option beam"
+	icon_state = "omnilaser"
+	nodamage = 1
+	damage = 5
+	damage_type = STAMINA
+	light_color = "#00CC33"
+
+	tracer_type = /obj/effect/projectile/tracer/pulse
+	muzzle_type = /obj/effect/projectile/muzzle/pulse
+	impact_type = /obj/effect/projectile/impact/pulse
+
+/obj/item/projectile/beam/final_option/on_hit(var/atom/impacted)
+	if(isliving(impacted))
+		var/mob/living/L = impacted
+		L.gib()
+
+	..()
+
+/obj/item/gunbox
+	name = "Modular Weapon Box"
+	desc = "A box containing a micro-fabricator capable of creating one of two different weapon selections. How fancy!"
+	icon = 'icons/obj/storage.dmi'
+	icon_state = "lockbox+l"
+	item_state = "syringe_kit"
+
+/obj/item/gunbox/attack_self(mob/user)
+	var/selection = input("Select weapon style.", "Modular Weapon Box", "Cancel") in list("X-01 Multiphase", "MWS-01 'Big Iron'", "Cancel")
```

Comment:
use an associative list of name = type
then just do var/obj/item (because why is it not doing items ever)/I = new list[choice] so it grabs the type
then do put in hands and qdel
no need for switch